tools/src/wyvern/target/corewyvernIL/expression/FieldSet.java

```diff
@@ -61,7 +83,42 @@ public ValueType typeCheck(TypeContext env) {
 
 	@Override
 	public Value interpret(EvalContext ctx) {
-		// TODO Auto-generated method stub
-		return null;
+		
+		// Evaluate object whose field is being set.
+		Value objExprVal = objectExpr.interpret(ctx);
+		if (!(objExprVal instanceof ObjectValue))
+			throw new RuntimeException("Runtime error: trying to set field of something which isn't an object.");
```

Comment:
A RuntimeException is probably OK here as this shouldn't happen once we get our typechecker working...although maybe we should eventually come up with a better architecture for run time errors too.